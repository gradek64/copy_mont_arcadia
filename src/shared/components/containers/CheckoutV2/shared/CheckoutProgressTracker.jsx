import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import ProgressTracker from '../../../common/ProgressTracker/ProgressTracker'
import { getProgressTrackerSteps } from '../../../../selectors/checkoutSelectors'

const mapStateToProps = (state) => ({
  steps: getProgressTrackerSteps(state),
})

const CheckoutProgressTracker = ({ steps }) => <ProgressTracker steps={steps} />

CheckoutProgressTracker.propTypes = {
  steps: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default connect(mapStateToProps)(CheckoutProgressTracker)
