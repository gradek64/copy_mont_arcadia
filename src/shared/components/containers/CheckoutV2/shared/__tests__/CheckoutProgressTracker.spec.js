import testComponentHelper from 'test/unit/helpers/test-component'
import CheckoutProgressTracker from '../CheckoutProgressTracker'
import trackerSteps from '../../../../../../../test/mocks/progressTrackerMocks'

describe('<CheckoutProgressTracker />', () => {
  const renderComponent = testComponentHelper(
    CheckoutProgressTracker.WrappedComponent
  )

  describe('@renders', () => {
    it('should render the correct steps for a new user', () => {
      const props = {
        steps: trackerSteps.NEW_CUSTOMER_STEPS,
      }
      expect(renderComponent(props).getTree()).toMatchSnapshot()
    })
    it('should render the correct steps for a returning user', () => {
      const props = {
        steps: trackerSteps.RETURNING_CUSTOMER_STEPS,
      }
      expect(renderComponent(props).getTree()).toMatchSnapshot()
    })
    it('should render the correct steps for a guest checkout user', () => {
      const props = {
        steps: trackerSteps.GUEST_CUSTOMER_STEPS,
      }
      expect(renderComponent(props).getTree()).toMatchSnapshot()
    })
  })
})
