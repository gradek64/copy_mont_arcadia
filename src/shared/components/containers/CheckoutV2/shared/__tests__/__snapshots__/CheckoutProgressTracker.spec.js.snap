// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutProgressTracker /> @renders should render the correct steps for a guest checkout user 1`] = `
<ProgressTracker
  steps={
    Array [
      Object {
        "active": false,
        "title": "Delivery",
        "url": "/guest/checkout/delivery",
      },
      Object {
        "active": true,
        "title": "Payment",
        "url": "/guest/checkout/payment",
      },
      Object {
        "active": false,
        "title": "Thank You",
      },
    ]
  }
/>
`;

exports[`<CheckoutProgressTracker /> @renders should render the correct steps for a new user 1`] = `
<ProgressTracker
  steps={
    Array [
      Object {
        "active": false,
        "title": "Delivery",
        "url": "/checkout/delivery",
      },
      Object {
        "active": true,
        "title": "Payment",
        "url": "/checkout/payment",
      },
      Object {
        "active": false,
        "title": "Thank You",
      },
    ]
  }
/>
`;

exports[`<CheckoutProgressTracker /> @renders should render the correct steps for a returning user 1`] = `
<ProgressTracker
  steps={
    Array [
      Object {
        "active": true,
        "title": "Delivery and Payment",
        "url": "/checkout/delivery-payment",
      },
      Object {
        "active": false,
        "title": "Thank You",
      },
    ]
  }
/>
`;
